ext/curl/interface.c

```diff
@@ -2106,16 +2106,169 @@ PHP_FUNCTION(curl_copy_handle)
 }
 /* }}} */
 
+static inline
+void php_curl_setopt_type_failure(zend_uchar exp, zval *arg, zend_bool nullable) { /* {{{ */
+	const char *exp_of = "be of type ";
+	const char *exp_type = "";
+
+	if (IS_CALLABLE == exp) {
+		exp_of = "be callable";
+	} else if (IS_ITERABLE == exp) {
+		exp_of = "be iterable";
+	} else {
+		exp_type = zend_get_type_by_const(exp);
+	}
+
+	zend_type_error("Argument 3 passed to curl_setopt() must %s%s, %s given",
+	                exp_of, exp_type, zend_zval_type_name(arg));
```

Comment:
In terms of developer experience, it would be really nice to see the option name in here, makes it a lot easier to trace back what went wrong.

As example:
 1. I see an error
 2. Line 10 indicates that I pass a string to `curl_setopt()`, but it expects an integer.
 3. I fix that error
 4. Line 11 indicates that I pass a integer to `curl_setopt()`, but it expects a boolean.
 5. Now I'm confused, because it looks like php is trolling me.

I propose something like this: `Argument 3 passed to curl_setopt() for option %s must %s%s, %s given`